devdoc/LDAPSecurityManager.md

```diff
@@ -0,0 +1,79 @@
+# LDAP Security Manager Init
+This document covers the main steps taken by the LDAP Security manager when initializing an AD via LDAP. An important
+distinction is that AD, Active Directory, is the store where permissions, users, password, resource, etc. are stored.
+LDAP is the protocol used to access the AD. The term LDAP is generally overused to refer to both.
+
+## LDAP Instances ##
+It is important to note that each service and or component using the LDAP security manager gets its
+own instance. This includes a unique connection pool (see below). This results in each loaded ESP service 
+in an ESP having its own instance. All operations described apply to each LDAP instance
+
+## Initialization Steps
+The following sections cover the main steps taken during initialization
+
+### Load Configuration
+The following items are loaded from the configuration
+
+#### AD Hosts
+The LDAP Security Manager supports using multiple ADs. The FDQN or IP address of each AD host is read from configuration
+data and stored internally. The source is a comma separated list stored in the _ldapAddress_ config value. Each entry
+is added to a pool of ADs
+
+**Note that all ADs are expected to use the same credentials and have the same configuration**
+
+#### AD Credentials
+AD credentials consist of a username and password. The LDAP security manager uses these to perform all operations
+on behalf of uses and components in the cluster. There are three potential sources for credentials.
+
+1. A Kubernetes secret. If the _ldapAdminSecretKey_ config value is set, but _ldapAdminVaultId_ is not (see 2) then
+the AD credentials are retrieve as Kubernetes secrets.
+2. If both _ldapAdminSecretKey_ and _ldapAdminVaultId_ config values are present, the AD credentials are retrieved
+from the vault.
+3. Hardcoded values from the _SystemCommonName_ and _systemPassword_ config values stored in the environment.xml file.
```

Comment:
Is the case for each "S" correct?  It's just inconsistent so I thought I'd ask.

Reply:
It is lower case 's' for systemCommonName.